server/genrest/testing.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DO NOT EDIT. This is an auto-generated file containing the REST handlers
+// for service #4: "Testing" (.google.showcase.v1beta1.Testing).
+
+package genrest
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/golang/protobuf/jsonpb"
+	genprotopb "github.com/googleapis/gapic-showcase/server/genproto"
+)
+
+// HandleCreateSession translates REST requests/responses on the wire to internal proto messages for CreateSession
+//    Generated for HTTP binding pattern: /v1beta1/sessions
```

Comment:
General question: Will we ever have to handle non-slash resource name separators here?

Reply:
I"m not sure. Do you have a reference about that? I haven't played much with resource names.

Regardless, URL paths have slash-separated components and variables are defined in terms of those, as per https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#path-template-syntax